src/wallet/spend.cpp

```diff
@@ -518,14 +520,27 @@ std::optional<SelectionResult> ChooseSelectionResult(const CWallet& wallet, cons
         results.push_back(*srd_result);
     }
 
-    if (results.size() == 0) {
+    if (results.empty()) {
         // No solution found
         return std::nullopt;
     }
 
+    std::vector<SelectionResult> eligible_results;
+    std::copy_if(results.begin(), results.end(), std::back_inserter(eligible_results), [coin_selection_params](const SelectionResult& result) {
+        const int weight = std::accumulate(result.GetInputSet().begin(), result.GetInputSet().end(), static_cast<int>(coin_selection_params.tx_noinputs_size * WITNESS_SCALE_FACTOR), [](int acc, const auto& coin) {
+            return acc + coin.input_bytes * WITNESS_SCALE_FACTOR;
+        });
+
+        return weight <= static_cast<int>(MAX_STANDARD_TX_WEIGHT);
+    });
+
+    if (eligible_results.empty()) {
```

Comment:
Perhaps we could store some “result hint” when we get into this if-block on a coin selection attempt that we could later use to inform the result message.

Reply:
Here's my attempt a3dece4e41de7bd5e118278abf09abcd2fcb4922.